crates/nu-protocol/src/span.rs

```diff
@@ -148,6 +148,28 @@ impl Span {
         }
     }
 
+    /// Converts row and column in a String to a Span, assuming bytes (1-based rows)
+    pub fn from_row_column(row: usize, col: usize, contents: &str) -> Span {
+        let mut cur_row = 1;
+        let mut cur_col = 1;
+
+        for (offset, curr_byte) in contents.bytes().enumerate() {
+            if curr_byte == b'\n' {
+                cur_row += 1;
+                cur_col = 1;
+            } else if cur_row >= row && cur_col >= col {
```

Comment:
The `if` to `else if` here fixes a bug where the span would point at one character before the actual problematic row/col